kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/security/auth/SimpleAclAuthorizer.java

```diff
@@ -138,12 +241,25 @@ protected PulsarService getPulsarService() {
 
     @Override
     public CompletableFuture<Boolean> canProduceAsync(KafkaPrincipal principal, Resource resource) {
+        checkArgument(resource.getResourceType() == ResourceType.TOPIC,
+                String.format("Expected resource type is TOPIC, but have [%s]", resource.getResourceType()));
         return authorize(principal, AuthAction.produce, resource);
     }
 
     @Override
     public CompletableFuture<Boolean> canConsumeAsync(KafkaPrincipal principal, Resource resource) {
+        checkArgument(resource.getResourceType() == ResourceType.TOPIC,
+                String.format("Expected resource type is TOPIC, but have [%s]", resource.getResourceType()));
         return authorize(principal, AuthAction.consume, resource);
     }
 
+    @Override
+    public CompletableFuture<Boolean> canCreateTopicAsync(KafkaPrincipal principal, Resource resource) {
+        checkArgument(resource.getResourceType() == ResourceType.TOPIC
+                || resource.getResourceType() == ResourceType.NAMESPACE,
+                String.format("Expected resource type is TOPIC or NAMESPACE, but have [%s]",
+                        resource.getResourceType()));
+        return authorize(principal, AuthAction.packages, resource);
+    }
+
```

Comment:
`AuthAction.packages` is not for this case. I think we should not authorize for the `CreateTopic` event. In Pulsar, the topic auto creation is only determined by the config or namespace policy.

Reply:
In kafka have authorized create topic event, don't we need this feature ?

Reply:
KoP uses Pulsar's authorization provider to grant permissions, so we should still use Pulsar's way. And Pulsar doesn't perform authorization for create topic event.

This permission is mainly for Kafka `AdminClient#createTopics` method, which is similar to Pulsar's admin operation (`PulsarAdmin#createPartitionedTopic`). In this case, we should only allow [roles with admin permission](https://pulsar.apache.org/docs/en/security-authorization/#administer-tenants.) to create topics.

Regarding to the auto topic creation case for Kafka client (not admin), we should still use configuration to control whether it's allowed, just like what Pulsar does.

Reply:
Agree, I will use Pulsar's way.